opencog/nlp/relex2logic/det.scm

```diff
@@ -0,0 +1,59 @@
+
+(define det
+    (BindLink
+        (VariableList
+            (TypedVariableLink
+                (VariableNode "$a-parse")
+                (TypeNode "ParseNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$noun")
+                (TypeNode "WordInstanceNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$det")
+                (TypeNode "WordInstanceNode")
+            )
+        )
+        (AndLink
+            (WordInstanceLink
+                (VariableNode "$noun")
+                (VariableNode "$a-parse")
+            )
+            (WordInstanceLink
+                (VariableNode "$det")
+                (VariableNode "$a-parse")
+            )
+            (EvaluationLink
+                (DefinedLinguisticRelationshipNode "_det")
+                (ListLink
+                    (VariableNode "$noun")
+                    (VariableNode "$det")
+                )
+            )
+            (InheritanceLink 
+                (VariableNode "$noun")
```

Comment:
The rule itself doesn't look second iteration to me.  It's only matching WordInstanceNode and ParseNode.

And even if the rule is for 2nd iteration, FC would add the result back into the focus set and see them in the 2nd step right?  Or did something change I am not aware that makes R2L only run 1 step?